python/cudf/cudf/_lib/column.pyx

```diff
@@ -501,22 +531,45 @@ cdef class Column:
             data_owner = owner.base_data
             mask_owner = mask_owner.base_mask
             base_size = owner.base_size
-
+        base_nbytes = base_size * dtype_itemsize
         if data_ptr:
             if data_owner is None:
                 data = as_device_buffer_like(
-                    rmm.DeviceBuffer(ptr=data_ptr,
-                                     size=(size+offset) * dtype.itemsize)
+                    rmm.DeviceBuffer(
+                        ptr=data_ptr,
+                        size=(size+offset) * dtype_itemsize
+                    ),
+                    exposed=False
                 )
+            elif (
+                # This is an optimizaion to avoid creating a new
+                # SpillableBuffer that represent the same memory
+                # as the owner.
+                column_owner and
+                isinstance(data_owner, SpillableBuffer) and
+                # We have to make sure that `data_owner` is already spill
+                # locked and that its pointer is the same as `data_ptr`
+                # _without_ exposing the buffer permanently.
+                not data_owner.spillable and
+                data_owner.get_ptr(spill_lock=SpillLock()) == data_ptr and
+                data_owner.size == base_nbytes
+            ):
+                data = data_owner
             else:
                 data = Buffer(
                     data=data_ptr,
-                    size=(base_size) * dtype.itemsize,
+                    size=base_nbytes,
                     owner=data_owner
                 )
+                if isinstance(data_owner, SpillableBuffer):
+                    # To prevent data_owner getting spilled, we attach an
+                    # SpillLock to data. This will make sure that data_owner
+                    # is unspillable as long as data is alive.
+                    data.spill_lock = SpillLock()
```

Comment:
Agree. It turns out that attaching a spill lock is not enough since the pointer of `data` can be exposed longer then the life of `data`. 
Since this is a very special case encountered in the `test_onehot.py` test, I now expose `data_owner`permanently.

Reply:
I think that makes sense. I'd rather err on the side of more spilling to get the code into a good place now rather than try to overoptimize. We can optimize further and improve more cases once we have a good starting point.